content/blog/2024-06-17-gsoc-multiboot2.mdx

```diff
@@ -0,0 +1,58 @@
+The BIOS's first goal is to run self test and initialization routines on hardware.
+As a result, interfaces are set up for the next software (the operating system) to understand the computer's resources without needing special drivers for each different hardware model.
+
+However, having the kernel deal with these low-level details directly is unnecessarily complex.
+This is where the bootloader comes in.
+After its initial routines, BIOS looks for bootable devices, such as a CD-ROM.
+If it finds one, the control is passed to the bootloader (a small portion of the executable code, found at the beginning of the device) which will then load the OS kernel into memory, and transfer control to it.
+
+There are multiple bootloader standards and protocols, some already supported by Unikraft.
+Nonetheless, there is still room to add other bootloader standards, such as Multiboot2, which is the focus of this project.
+
+Multiboot2, an enhanced version of Multiboot, solves some of the limitations of the original protocol, such as the lack of support for 64-bit UEFI systems.
+It also provides a more accurate and detailed communication during the bootstrapping process and offers a standardized tag system for easier configuration.
+
+### About Me
+
+My name is [Maria Pană](https://github.com/mariapana) and I am wrapping up my second year of studies at [University POLITEHNICA of Bucharest](https://upb.ro/en/).
+I am passionate about operating systems and low-level programming, which makes me thrilled to be part of the Unikraft community and work on this project during GSoC'24!
+
+### Progress
+
+Making changes or improvements to software generally involves three phases: understanding the existing code, making the necessary changes, and testing the new code.
+This week I focused on the first phase, by revisting the [GNU Multiboot2 specification](https://www.gnu.org/software/grub/manual/multiboot2/multiboot.html) and the notes I made prior to applying to GSoC.
+I also took a look at a previous attempt to integrate Multiboot2 in Unikraft and compared it to the original specification.
+
+Following the advice of my mentors, I then learned how to build GRUB and Multiboot2 images, and built GRUB from source.
+This helped me understand the process of creating a bootable image for a custom kernel and, by attaching GDB to the built GRUB, I prepared my "weapon" for future debugging sessions.
+
+Using the insight gained from these activities, I took a look at Unikraft's `mkukimg` script for creating bootable images, which will need updating to support Multiboot2.
+
+Lastly, I researched Unikraft's current Multiboot implementation, as well as how other projects handled Multiboot2 integration to gain a wider perspective on the overall flow of the process moving forward.
+
+### Next Steps
+
+Up next, I aim to get started on the actual implementation of Multiboot2 in Unikraft by adding the standard structures and files needed for the protocol.
```

Comment:
Can you give some examples of what are the Multiboot2 tags and/or the parts of the structure that Unikraft will use (and how it will make use of them).